dsl/pace/dsl/dace/sdfg_opt_passes.py

```diff
@@ -0,0 +1,126 @@
+import collections
+from typing import Dict, List
+
+import dace
+from dace import data, subsets
+
+
+def make_local_memory_transient(sdfg: dace.SDFG):
+    """
+    Detect memory used only locally to a nested SDFG and turns it into
+    transient (GPU shared & scoped to the nested SDFG) when it has been
+    declared has global.
+    This will reduce memory pressure.
+
+    As a custom optimization pass, this is to be inserted after the
+    generic optimization call to sdfg.simplify(...)
+    """
```

Comment:
This is so much clearer (code too), thank you! I had misunderstood what this was doing before.